docs/guide/headless-cms.md

```diff
@@ -0,0 +1,177 @@
+    "slug": "/",
+    // ... additional properties
+    "_url": "http://localhost:3000/",
+    "_ancestors": [],
+    // Child pages 👇
+    "_children": [
+        {
+            "_id": "ckhfen8ls0005mq9k8p9izkjt",
+            "title": "Blog",
+            "slug": "/blog",
+            // ... additional properties
+            "_url": "http://localhost:3000/blog",
+            "_children": []
+        },
+        {
+            "_id": "ckhurshqd00088v9k6pfnqpjz",
+            "title": "About us",
+            "slug": "/about-us",
+            // ... additional properties
+            "_url": "http://localhost:3000/about-us",
+            "_children": []
+        }
+    ]
+}
+```
+
+The home page object is returned with a `_children` array with the first-level pages.
+
+**We can request this data as a flat array** by adding the `?flat=1` query parameter.
+
+```
+GET https://example.rocks/api/v1/@apostrophecms/page?flat=1&apikey=000apikey123
+```
+
+**And expand the results to include all pages, regardless of page tree depth** by using the `?all=1` query parameter. This also works with the nested object format.
+
+```
+GET https://example.rocks/api/v1/@apostrophecms/page?flat=1&all=1apikey=000apikey123
+```
+
+## Getting rendered HTML in responses
+
+Using an object of JSON-like data from a REST API is very normal from a headless CMS. Traditionally the application's "head" (such as a single page Vue.js, React, or Angular application) would use that structured data to display product, event, or similar content information. It is less common for a headless CMS to deliver page content that is tied to particular HTML markup and layout. More often, landing page text and images might be coded directly into the app code or otherwise not be from the same data source.
+
+Apostrophe support both approaches. The in-context editing features and [widget areas](/guide/areas-and-widgets.md) give editors more control over content, but in it's default form areas are fairly incomprehensible data objects. It is when they are rendered with the widget template files that they are fully realized.
+
+**In any Apostrophe REST API single-document `GET` ("get one") request, include the `?render-areas=1` query parameter.** Each area field in the response will have a `_rendered` property with a string of HTML instead of the `items` array of widget objects.
+
+```
+GET https://example.rocks/api/v1/@apostrophecms/page/903y5n76e8j:en:published?apikey=000apikey123&render-areas=1
+```
+
+If we wanted to get the full page as HTML rendered using the page templates, we could make a `GET` request to the page URL (basically what browsers do).
+
+```
+GET https://example.rocks/about-us
+```
+
+ More likely we would only want the unique page content (without the `head` tag and outer wrapper elements), which we can get by adding the `aposRefresh=1` query parameter. It will include only the HTML markup for the page template.
+
+```
+GET https://example.rocks/about-us?aposRefresh=1
+```
+
```

Comment:
Can we add a note that describes examples of when you might use these different approaches?